skimage/color/colorconv.py

```diff
@@ -1455,3 +1460,38 @@ def _prepare_lab_array(arr):
     if shape[-1] < 3:
         raise ValueError('Input array has less than 3 color channels')
     return dtype.img_as_float(arr, force_copy=True)
+
+
+def colormap_image(img, cmap):
+    """Draw a grayscale image by using the given colormap.
+
+    Parameters
+    ----------
+    img : (M, N) array
+        The input image.
+    cmap : string or :class:`matplotlib.colors.Colormap` instance.
+        The colormap to use
+
+    Returns
+    -------
+    out: (M, N, 3) array
+        The image drawn with the specified colormap in RGB format.
+
+    Examples
+    --------
+    >>> from skimage import data, color
+    >>> from skimage.util.colormap import viridis
+    >>> out = color.colormap_image(data.camera(), viridis)
+    """
+
+    cmap = plt.get_cmap(cmap)
+    img = img_as_ubyte(img)
+    gray_colors = np.arange(0, 256, dtype=np.int)
```

Comment:
I think it would be better to let user decide the number of color levels (add `num_levels`/`precision` to the function API). For this, to use `0, 1, ...`, not `0, 256, ...`: `Typically Colormap instances are used to convert data values (floats) from the interval [0, 1]` (http://matplotlib.org/api/colors_api.html#matplotlib.colors.Colormap).